File: app/src/main/java/com/testapp/testapp/model/utils/DateFormatter.java
package com.testapp.testapp.model.utils;

import java.text.SimpleDateFormat;
import java.util.Date;

/**
 * Created by troll on 05.11.2017.
 */

public class DateFormatter {

    /**
     * Transform date to format for Foursquare 'v' parameter in GET requests
     * */
    public static String getCurrentDateInApiVersionFormat(){
        Date currentDate = new Date();
        SimpleDateFormat simpleDateFormat = new SimpleDateFormat("yyyyMMdd");
        return simpleDateFormat.format(currentDate);
    }
}

File: app/src/main/java/com/testapp/testapp/kotlin/presenter/utils/DateFormatter.kt
package com.testapp.testapp.kotlin.presenter.utils

import java.text.SimpleDateFormat
import java.util.*

class DateFormatter {

    companion object {
        /**
         * Transform date to format for Foursquare 'v' parameter in GET requests
         * */
        fun getCurrentDateInApiVersionFormat(): String {
            val currentDate = Date()
            val simpleDateFormat = SimpleDateFormat("yyyyMMdd")
            return simpleDateFormat.format(currentDate)
        }
    }
}